tuf/snapshot.py

```diff
@@ -0,0 +1,258 @@
+import abc
+import os
+import logging
+
+from securesystemslib import formats as sslib_formats
+
+from tuf import formats
+from tuf import roledb
+from tuf import repository_lib as repolib
+
+# See 'log.py' to learn how logging is handled in TUF.
+logger = logging.getLogger(__name__)
+
+# The extension of TUF metadata.
+METADATA_EXTENSION = '.json'
+
+# The metadata filenames of the top-level roles.
+ROOT_FILENAME = 'root' + METADATA_EXTENSION
+TARGETS_FILENAME = 'targets' + METADATA_EXTENSION
+SNAPSHOT_FILENAME = 'snapshot' + METADATA_EXTENSION
+TIMESTAMP_FILENAME = 'timestamp' + METADATA_EXTENSION
+
+class SnapshotInterface():
+  """
+  <Purpose>
+  Defines an interface for abstract snapshot metadata operations
+  to be implemented for a variety of snapshot creation methods,
+  including the classic manifest snapshot metadata and snapshot
+  merkle trees
+  """
+
+  __metaclass__ = abc.ABCMeta
+
+  @abc.abstractmethod
+  def add_to_snapshot(self, rolename : str):
+    """
+    <Purpose>
+     Indicate to the snapshot interface that 'rolename'
+     should be included in the next snapshot generation.
+
+    <Arguments>
+     rolename:
+      The name of the role to be added
+
+    <Exceptions>
+     TODO
+
+    <Returns>
+     None
+    """
+    raise NotImplementedError # pragma: no cover
+
+
+
+
+  @abc.abstractmethod
+  def remove_from_snapshot(self, rolename : str):
+    """
+    <Purpose>
+     Indicate to the snapshot interface that 'rolename'
+     should be removed from the next snapshot generation.
+
+    <Arguments>
+     rolename:
+      The name of the role to be removed
+
+    <Exceptions>
+     TODO
+
+    <Returns>
+     None
+    """
+    raise NotImplementedError # pragma: no cover
+
+
+
+
+  @abc.abstractmethod
+  def generate_snapshot_metadata(self, metadata_directory, version, expiration_date,
+    storage_backend, consistent_snapshot=False,
+    repository_name='default', use_length=False, use_hashes=False):
+    """
+    <Purpose>
+		 Create the snapshot metadata
+
+    <Arguments>
```

Comment:
A lot of these seem to me like the belong to the class (and specifically, should correspond to *different* classes in some cases). At least the following:

- consistent_snapshot
- use_length
- use_hashes